### PR TITLE
Added keystore type and truststore type to rabbit properties

### PR DESCRIPTION
**Pull request by martingreber**
Added autoconfig support for properties:
spring.rabbitmq.ssl.key-store-type
spring.rabbitmq.ssl.trust-store-type

This simplifies configuration if you only use JKS key/trust stores. Cheers!

### CODE REVIEW

**Comment by martingreber** [comment]
All done :)

File: spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitProperties.java [hunk 1]
```diff
@@ -315,6 +316,11 @@ public Template getTemplate() {
 		private String keyStore;
 
 		/**
+		 * Set the key store type. See {@link RabbitConnectionFactoryBean#getKeyStoreType()}
```
Comment (wilkinsona) [inline spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitProperties.java]: Could you remove the second sentence please? We don't use Javadoc links in the field javadoc for configuration properties as it messes up the auto-generated metadata used by your IDE.

File: spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitProperties.java [hunk 2]
```diff
@@ -325,6 +331,11 @@ public Template getTemplate() {
 		private String trustStore;
 
 		/**
+		 * Set the trust store type. See {@link RabbitConnectionFactoryBean#getTrustStoreType()}
```
Comment (wilkinsona) [inline spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitProperties.java]: Could you remove the second sentence please? We don't use Javadoc links in the field javadoc for configuration properties as it messes up the auto-generated metadata used by your IDE.

File: spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitProperties.java (hunk 1 above)
Comment (martingreber) [reply spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitProperties.java]: Cheers Andy. Good point by the way. Done!

**Comment by wilkinsona** [comment]
Very nice. Thanks again.

**Pull request closed without merging**